test/utils/permit2.test.ts

```diff
@@ -0,0 +1,65 @@
+import { expect } from 'chai'
+import { BigNumber, utils, Wallet } from 'ethers'
+import { PermitSingle } from '@uniswap/permit2-sdk'
+import { defaultAbiCoder } from 'ethers/lib/utils'
+import { encodePermit } from '../../src/utils/permit2'
+import { RoutePlanner } from '../../src/utils/routerCommands'
+import { USDC } from './uniswapData'
+import { generatePermitSignature, makePermit } from './permit2'
+
+const PERMIT_STRUCT =
+  '((address token,uint160 amount,uint48 expiration,uint48 nonce) details, address spender, uint256 sigDeadline)'
+
+// note: these tests aren't testing much but registering calldata to interop file
+// for use in forge fork tests
+describe('Permit2', () => {
+  const wallet = new Wallet(utils.zeroPad('0x1234', 32))
+
+  describe('v2', () => {
+    it('does not sanitize a normal permit', async () => {
+      const inputUSDC = utils.parseUnits('1000', 6).toString()
+      const permit = makePermit(USDC.address, inputUSDC)
+      const signature = await generatePermitSignature(permit, wallet, 1)
+      const sanitized = getSanitizedSignature(permit, signature)
+      expect(sanitized).to.equal(signature)
+    })
+
+    it('does not sanitize a triple length permit', async () => {
+      const inputUSDC = utils.parseUnits('1000', 6).toString()
+      const permit = makePermit(USDC.address, inputUSDC)
+      const signature = await generatePermitSignature(permit, wallet, 1)
+      const multisigSignature = signature + signature.slice(2) + signature.slice(2)
+      const sanitized = getSanitizedSignature(permit, multisigSignature)
+      expect(sanitized).to.equal(multisigSignature)
+    })
+
+    it('does not sanitize a short permit', async () => {
+      const inputUSDC = utils.parseUnits('1000', 6).toString()
+      const permit = makePermit(USDC.address, inputUSDC)
+      const tinySignature = '0x12341234132412341344'
+      const sanitized = getSanitizedSignature(permit, tinySignature)
+      expect(sanitized).to.equal(tinySignature)
+    })
+
+    it('sanitizes a malformed permit', async () => {
```

Comment:
should test smart contract signature also?

Reply:
the long one is basically a 1271 sig